### PR TITLE
fix(tools): move-packages generator now updates moved package's build:local script

### PR DESCRIPTION
**Pull request by TristanWatanabe**
## Changes

- `move-packages` generator is now updating the `build:local` script of each package that gets moved to point to the new relative path of the moved package
- Adds an `AssertedSchema` interface and refactors some functions to use it to simplify implementation.

## Related Issue(s)

Follow-up to #22664 and [comment](https://github.com/microsoft/fluentui/pull/22664#pullrequestreview-954815205)

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/35e679c49aab011abc04abadf6b34bbd7bd09188'>35e679c49aab011abc04abadf6b34bbd7bd09188</a></sub>
<div id="bundleSizeComment9423"/>

File: tools/generators/move-packages/index.ts [hunk 1]
```diff
@@ -151,7 +152,7 @@ function updateStorybookTypeImport(tree: Tree, schema: MovePackagesGeneratorSche
 function updateCodeOwners(tree: Tree, schema: MovePackagesGeneratorSchema) {
   const { name, destination } = schema;
   if (name) {
-    const pkgName = name.split('/')[1];
+    const pkgName = name.split('/').slice(-1).join('/');
```
Comment (Hotell) [inline tools/generators/move-packages/index.ts:155]: shouldn't this be the same as on line 173 ? 
`split('/').slice(0, -1).join('/');`

Comment (Hotell) [reply tools/generators/move-packages/index.ts:155]: ah nevermind :D it's package name

File: tools/generators/move-packages/index.ts [hunk 2]
```diff
@@ -178,3 +179,18 @@ function updateApiExtractor(tree: Tree, schema: MovePackagesGeneratorSchema) {
 
   writeJson(tree, apiExtractorLocalPath, apiExtractorLocal);
 }
+
+function updateBuildLocalScript(tree: Tree, schema: MovePackagesGeneratorSchema) {
+  const { name, destination } = schema;
```
Comment (Hotell) [inline tools/generators/move-packages/index.ts:184]: this implementation can be simplified to following:

```ts
function updateBuildLocalScript(tree: Tree, schema: MovePackagesGeneratorSchema) {
  const project = getProjectConfig(tree, { packageName: schema.name! });

  updateJson<PackageJson>(tree, project.paths.packageJson, json => {
    // 💡 this is a defensive programming, enforced by TS types 
    json.scripts = json.scripts ?? {};
    const originalScript = json.scripts['build:local'];
    const updatedScript = originalScript.replace(project.normalizedPkgName, schema.destination);
    json.scripts['build:local'] = updatedScript;
    return json;
  });
}
```


**Out of topic:**

I'd suggest to introduce something like 

```ts
interface AssertedSchema extends MovePackagesGeneratorSchema {
  name: string;
}
```

and use that interface for those atomic functions as `name` is required for them to make it work am I right ?

```ts
function updateBuildLocalScript(tree: Tree, schema: AssertedSchema) {
....
```

File: tools/generators/move-packages/index.ts (hunk 1 above)
Comment (TristanWatanabe) [reply tools/generators/move-packages/index.ts]: I just wanted the normalized package name in this case, changed the approach and used `getProjectConfig` instead to get that

File: tools/generators/move-packages/index.ts (hunk 2 above)
Comment (TristanWatanabe) [reply tools/generators/move-packages/index.ts]: Made the refactor of the `updateBuildLocalScript` function and added an `AssertedSchema` like you suggested in this commit [2d9cd3c](https://github.com/microsoft/fluentui/pull/22667/commits/2d9cd3c0a057df4f763c3842565c93679c2bbcde). I refactored some of the other functions to use the `AssertedSchema` so sorry if the diffs are slightly messy 😅

**Review by Hotell — APPROVED** [review]
🙌

**Pull request merged**